pygmt/src/xyz2grd.py

```diff
@@ -51,11 +52,13 @@ def xyz2grd(data, **kwargs):
 
         - :class:`xarray.DataArray`: if ``outgrid`` is not set
         - None if ``outgrid`` is set (grid output will be stored in file set by
-          ``outgrid``)```
+          ``outgrid``)
     """
     with GMTTempFile(suffix=".nc") as tmpfile:
         with Session() as lib:
-            file_context = lib.virtualfile_from_data(check_kind="vector", data=data)
+            file_context = lib.virtualfile_from_data(
+                check_kind="vector", data=data, x=x, y=y, z=z, required_z=True
+            )
             with file_context as infile:
                 if "G" not in kwargs.keys():  # if outgrid is unset, output to tempfile
                     kwargs.update({"G": tmpfile.name})
```

Comment:
Could you combine L66:L67 below to use `arg_str = " ".join([infile, build_arg_string(kwargs)])`? Makes it consistent with other functions like `sph2grd`.